### ISSUE TITLE
Static NSRegularExpression initializations

### ISSUE THREAD

**Issue by glenyi** [mdiep/MMMarkdown#106]
For large markdown files, forming the regex can be slow and doesn't need to be initialized more than once. The following adds a large performance boost in MMSpanParser:

```
    static NSRegularExpression *regex;
    static dispatch_once_t onceToken;
    dispatch_once(&onceToken, ^{
        regex = [NSRegularExpression regularExpressionWithPattern:@"^(\\w+)://" options:0 error:nil];
    });
```

Same with the email link scanner:
```
    static NSRegularExpression *regex;
    static dispatch_once_t onceToken;
    dispatch_once(&onceToken, ^{
        regex = [NSRegularExpression regularExpressionWithPattern:@"^[-._0-9\\p{L}]+@[-\\p{L}0-9][-.\\p{L}0-9]*\\.\\p{L}+$"
                                                          options:NSRegularExpressionCaseInsensitive
                                                            error:nil];
    });
```

**Comment by mdiep** [comment]
Care to open a pull request?

**Comment by glenyi** [comment]
Done :)

**Comment by mdiep** [comment]
Fixed by #107.
